### PR TITLE
Shorthands

### PR DESCRIPTION
**Pull request by talh78**

### CODE REVIEW

File: src/shorthands/openers/background-shorthand.ts [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    prop: 'background-origin',
+    dataType: backgroundOriginDataType,
+    opener: splitShorthandOpener(['background-origin', 'background-clip']),
+  },
+  { prop: 'background-clip',       dataType: backgroundClipDataType },
+];
+const finalBgLayerShorthandParts: ShorthandPart<string>[] = [{
+  prop: 'background-color',
+  dataType: backgroundColorDataType,
+} as ShorthandPart<string>].concat(bgLayerShorthandParts);
```
Comment (danieleww) [inline src/shorthands/openers/background-shorthand.ts:37]: no need for ```as ShorthandPart<string>```

File: src/shorthands/shorthand-parser-utils.ts [hunk 2]
```diff
@@ -0,0 +1,338 @@
...
+};
+
+export const singleKeywordShorthandOpener = <T extends string>(
+  keywordValueMap: Record<string, Record<T, string>>,
+): ShorthandOpenerInner<T> => astNodes => {
+  let opened: OpenedShorthand<string> = {};
+  const keywordValues = keywordValueMap[astNodes[0].value.text];
+  const props = Object.keys(keywordValues) as T[];
+  for (const prop of props) {
+    opened[prop] = getDefaultAst(keywordValues[prop as T]);
```
Comment (danieleww) [inline src/shorthands/shorthand-parser-utils.ts:171]: no need for ```as T```

File: src/shorthands/openers/font-shorthand.ts [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+        prefixEndIndex = i;
+        break;
+      }
+    }
+    if (prefixEndIndex !== -1) {
+      opened = {
+        ...openFontShorthandPrefixInner(
+          astNodes.slice(0, prefixEndIndex),
+          api,
+        ) as Record<FontPrefixes, EvaluatedAst>,
```
Comment (danieleww) [inline src/shorthands/openers/font-shorthand.ts:65]: no need for ```as Record<... ```

File: src/shorthands/openers/font-shorthand.ts [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+        ...openFontShorthandSuffixInner(
+          astNodes.slice(prefixEndIndex),
+          api,
+        ) as Record<FontSuffixes, EvaluatedAst>,
```
Comment (danieleww) [inline src/shorthands/openers/font-shorthand.ts:69]: same

**Review by danieleww — APPROVED** [review]

**Review by barak007 — CHANGES_REQUESTED** [review]
This PR should be extracted to a separate repo. 
The tokenizer abilities can stay but any shorthand expansion must go.

**Comment by talh78** [comment]
Code moved to a new repository [here](https://github.com/wixplosives/shorthands-opener)

**Pull request closed without merging**